kafka/adminapi.go

```diff
@@ -312,6 +333,222 @@ func (c ConfigResourceResult) String() string {
+	// DESCRIBE_CONFIGS operation
+	AclOperationDescribeConfigs = AclOperation(C.RD_KAFKA_ACL_OPERATION_DESCRIBE_CONFIGS)
+	// ALTER_CONFIGS operation
+	AclOperationAlterConfigs = AclOperation(C.RD_KAFKA_ACL_OPERATION_ALTER_CONFIGS)
+	// IDEMPOTENT_WRITE operation
+	AclOperationIdempotentWrite = AclOperation(C.RD_KAFKA_ACL_OPERATION_IDEMPOTENT_WRITE)
+)
+
+// String returns the human-readable representation of an AclOperation
+func (o AclOperation) String() string {
+	return C.GoString(C.rd_kafka_AclOperation_name(C.rd_kafka_AclOperation_t(o)))
+}
+
+// AclOperationFromString translates a ACL operation name to
+// a AclOperation value.
+func AclOperationFromString(aclOperationString string) (AclOperation, error) {
+	switch strings.ToUpper(aclOperationString) {
+	case "ANY":
+		return AclOperationAny, nil
+	case "ALL":
+		return AclOperationAll, nil
+	case "READ":
+		return AclOperationRead, nil
+	case "WRITE":
+		return AclOperationWrite, nil
+	case "CREATE":
+		return AclOperationCreate, nil
+	case "DELETE":
+		return AclOperationDelete, nil
+	case "ALTER":
+		return AclOperationAlter, nil
+	case "DESCRIBE":
+		return AclOperationDescribe, nil
+	case "CLUSTER_ACTION":
+		return AclOperationClusterAction, nil
+	case "DESCRIBE_CONFIGS":
+		return AclOperationDescribeConfigs, nil
+	case "ALTER_CONFIGS":
+		return AclOperationAlterConfigs, nil
+	case "IDEMPOTENT_WRITE":
+		return AclOperationIdempotentWrite, nil
+	default:
+		return AclOperationUnknown, NewError(ErrInvalidArg, "Unknown ACL operation", false)
+	}
+}
+
+// Enumerates the different types of ACL permission types.
+type AclPermissionType int
+
+const (
+	// Unknown
+	AclPermissionTypeUnknown = AclPermissionType(C.RD_KAFKA_ACL_PERMISSION_TYPE_UNKNOWN)
+	// In a filter, matches any AclPermissionType
+	AclPermissionTypeAny = AclPermissionType(C.RD_KAFKA_ACL_PERMISSION_TYPE_ANY)
+	// Disallows access
+	AclPermissionTypeDeny = AclPermissionType(C.RD_KAFKA_ACL_PERMISSION_TYPE_DENY)
+	// Grants access
+	AclPermissionTypeAllow = AclPermissionType(C.RD_KAFKA_ACL_PERMISSION_TYPE_ALLOW)
+)
+
+// String returns the human-readable representation of an AclPermissionType
+func (o AclPermissionType) String() string {
+	return C.GoString(C.rd_kafka_AclPermissionType_name(C.rd_kafka_AclPermissionType_t(o)))
+}
+
+// AclPermissionTypeFromString translates a ACL permission type name to
+// a AclPermissionType value.
+func AclPermissionTypeFromString(aclPermissionTypeString string) (AclPermissionType, error) {
+	switch strings.ToUpper(aclPermissionTypeString) {
+	case "ANY":
+		return AclPermissionTypeAny, nil
+	case "DENY":
+		return AclPermissionTypeDeny, nil
+	case "ALLOW":
+		return AclPermissionTypeAllow, nil
+	default:
+		return AclPermissionTypeUnknown, NewError(ErrInvalidArg, "Unknown ACL permission type", false)
+	}
+}
+
+// Represents an ACL binding that specify the operation and permission type for a specific principal
```

Comment:
Believe docstrings must begin with the typename, e.g 'AclBinding specify the operation and permi...'